examples/tests/valid/bucket_events.w

```diff
@@ -3,24 +3,24 @@ bring cloud;
 let other = new cloud.Bucket() as "other";
 let b = new cloud.Bucket() as "b";
 
-b.onDelete(inflight (key: str) => {
+b.onDelete(inflight (key: str) -> void {
     log("deleted ${key}");
 });
 
-b.onUpdate(inflight (key: str) => {
+b.onUpdate(inflight (key: str) -> void {
     log("updated ${key}");
 });
 
-b.onCreate(inflight (key:str) => {
+b.onCreate(inflight (key:str) -> void {
     log("created ${key}");
 });
 
-b.onEvent(inflight (key: str) => {   
+b.onEvent(inflight (key: str) -> void {
```

Comment:
A major downside of this syntax is that if we want to support inference of return types (we already sort of have that with `void`), the syntax breaks:

```js
b.onEvent((key: str) {
  // ...
});
```

Right? That looks odd.

Reply:
I think this is one reason why inferring return types might not be desirable (I've also alluded to some other reasons [here](https://github.com/winglang/wing/issues/2146#issuecomment-1514905983))

Reply:
I tend to agree that for *methods* we should require a return type annotation (for the reasons you state in your comment), but for *closures*, it's a huge ergonomic win.

So I think that a syntax that doesn't really allow us to add return type inference for closures is not ideal...

Reply:
One solution to this could be to add `fn` to the syntax for anonymous functions. For example, I think this might look reasonable:

```js
b.onEvent(fn(key) {
  // ...
});
```

With trailing closures, the `fn` could be omitted:

```js
b.onEvent { key in
  // ...
};
```